app/screens/edit_server/form.tsx

```diff
@@ -0,0 +1,198 @@
+import {t} from '@i18n';
+import {buttonBackgroundStyle, buttonTextStyle} from '@utils/buttonStyles';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+import {removeProtocol, stripTrailingSlashes} from '@utils/url';
+
+type Props = {
+    buttonDisabled: boolean;
+    connecting: boolean;
+    displayName?: string;
+    displayNameError?: string;
+    handleUpdate: () => void;
+    handleDisplayNameTextChanged: (text: string) => void;
+    keyboardAwareRef: MutableRefObject<KeyboardAwareScrollView | undefined>;
+    serverUrl: string;
+    theme: Theme;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    formContainer: {
+        alignItems: 'center',
+        maxWidth: 600,
+        width: '100%',
+        paddingHorizontal: 20,
+    },
+    fullWidth: {
+        width: '100%',
+    },
+    error: {
+        marginBottom: 18,
+    },
+    chooseText: {
+        alignSelf: 'flex-start',
+        color: changeOpacity(theme.centerChannelColor, 0.64),
+        marginTop: 8,
+        ...typography('Body', 75, 'Regular'),
+    },
+    connectButton: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        width: '100%',
+        marginTop: 32,
+        marginLeft: 20,
+        marginRight: 20,
+        padding: 15,
+    },
+    loadingContainerStyle: {
+        marginRight: 10,
+        padding: 0,
+        top: -2,
+    },
+    loading: {
+        height: 20,
+        width: 20,
+    },
+}));
+
+const EditServerForm = ({
+    buttonDisabled,
+    connecting,
+    displayName = '',
+    displayNameError,
+    handleUpdate,
+    handleDisplayNameTextChanged,
+    keyboardAwareRef,
+    serverUrl,
+    theme,
+}: Props) => {
+    const {formatMessage} = useIntl();
+    const isTablet = useIsTablet();
+    const dimensions = useWindowDimensions();
+    const displayNameRef = useRef<FloatingTextInputRef>(null);
+    const styles = getStyleSheet(theme);
+
+    const focus = () => {
+        if (Platform.OS === 'ios') {
+            let offsetY = 160;
+            if (isTablet) {
+                const {width, height} = dimensions;
+                const isLandscape = width > height;
+                offsetY = isLandscape ? 230 : 100;
+            }
+            requestAnimationFrame(() => {
+                keyboardAwareRef.current?.scrollToPosition(0, offsetY);
+            });
+        }
+    };
+
+    const onBlur = useCallback(() => {
+        if (Platform.OS === 'ios') {
+            const reset = !displayNameRef.current?.isFocused();
+            if (reset) {
+                keyboardAwareRef.current?.scrollToPosition(0, 0);
+            }
+        }
+    }, []);
+
+    const onUpdate = useCallback(() => {
+        Keyboard.dismiss();
+        handleUpdate();
+    }, [buttonDisabled, connecting, displayName, theme]);
+
+    const onFocus = useCallback(() => {
+        focus();
+    }, [dimensions]);
```

Comment:
focus also depends on isTablet.

Inferring dependencies is brittle and error prone, so I would prefer to depend on focus and make focus to use a useCallback. (0/5)

Reply:
yeah I'm adding the dependency to avoid issues although when the tablet goes into split view it also means that the dimensions change, but hey let's add it anyway.